OMEdit/OMEditLIB/OMS/ElementPropertiesDialog.cpp

```diff
@@ -403,3 +405,91 @@ void ElementPropertiesDialog::updateProperties()
   // accept the dialog
   accept();
 }
+
+/*
+ * event filter for mParameterLineEdits and mInputLineEdit
+ * to detect the focus out event, and update
+ * the default start values from modeldesctiption.xml
+ */
+bool ElementPropertiesDialog::eventFilter(QObject *pObject, QEvent *pEvent)
+{
+  QLineEdit *pLineEdit = qobject_cast<QLineEdit*>(pObject);
+
+  if (pLineEdit && pEvent->type() == QEvent::FocusOut) {
+    if (pLineEdit->text().isEmpty()) {
+      // search the lineEdit index in parameters or inputs
+      int parameterIndex = mParameterLineEdits.indexOf(pLineEdit);
+      int inputIndex = mInputLineEdits.indexOf(pLineEdit);
+
+      if (parameterIndex != -1) {
+        QString pParameterLabelText = mParameterLabels.at(parameterIndex)->text();
```

Comment:
Wrong naming style. Should be `parameterLabelText` instead of `pParameterLabelText`. You should also take care of index out of bound exception. Probably enough to check if index is valid before accessing it.

Reply:
`if (parameterIndex != -1) ` , does this condition, not check if it is a valid index or not ?

Reply:
OK you are using `QList::indexOf` then its enough.